framework/markdown_test.go

```diff
@@ -0,0 +1,257 @@
+	}
+}
+
+func TestGetComponentTitle(t *testing.T) {
+	tests := []struct {
+		assessmentPlan oscalTypes.AssessmentPlan
+		expected       string
+		hasError       bool
+	}{
+		{
+			assessmentPlan: oscalTypes.AssessmentPlan{
+				LocalDefinitions: &oscalTypes.LocalDefinitions{
+					Components: &[]oscalTypes.SystemComponent{
+						{
+							Title: "Component Title",
+						},
+					},
+				},
+			},
+			expected: "Component Title",
+			hasError: false,
+		},
+		{
+			assessmentPlan: oscalTypes.AssessmentPlan{
+				LocalDefinitions: &oscalTypes.LocalDefinitions{
+					Components: &[]oscalTypes.SystemComponent{},
+				},
+			},
+			expected: "",
+			hasError: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.expected, func(t *testing.T) {
+			result, err := getComponentTitle(test.assessmentPlan)
+			if test.hasError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.expected, result)
+			}
+		})
+	}
+}
+
+func TestExtractControlId(t *testing.T) {
+	test := struct {
+		targetId string
+		expected string
+	}{
+		targetId: "control-1_smt",
+		expected: "control-1",
+	}
+	result := extractControlId(test.targetId)
+	require.Equal(t, test.expected, result)
+}
+
+func TestExtractRuleId(t *testing.T) {
+	// Setup mock data
+	ob := oscalTypes.Observation{
+		UUID: "1234-uuid",
+		Props: &[]oscalTypes.Property{
+			{
+				Name:  "assessment-rule-id",
+				Value: "rule-1",
+			},
+		},
+	}
+	tests := []struct {
+		observation     oscalTypes.Observation
+		observationUuid string
+		expected        string
+	}{
+		{
+			observation:     ob,
+			observationUuid: "1234-uuid",
+			expected:        "rule-1",
+		},
+		{
+			observation:     ob,
+			observationUuid: "wrong-uuid",
+			expected:        "",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.observationUuid, func(t *testing.T) {
+			result := extractRuleId(test.observation, test.observationUuid)
+			require.Equal(t, test.expected, result)
+		})
+	}
+}
+
+func TestCreateTemplateValues(t *testing.T) {
+	catalog := oscalTypes.Catalog{
+		Metadata: oscalTypes.Metadata{
+			Title: "Catalog Title",
+		},
+	}
+	assessmentPlan := oscalTypes.AssessmentPlan{
+		LocalDefinitions: &oscalTypes.LocalDefinitions{
+			Components: &[]oscalTypes.SystemComponent{
+				{Title: "Component Title"},
+			},
+		},
+	}
+	assessmentResults := oscalTypes.AssessmentResults{}
+
+	test := struct {
+		catalog           oscalTypes.Catalog
+		assessmentPlan    oscalTypes.AssessmentPlan
+		assessmentResults oscalTypes.AssessmentResults
+		expected          *TemplateValues
+	}{
+		catalog:           catalog,
+		assessmentPlan:    assessmentPlan,
+		assessmentResults: assessmentResults,
+		expected: &TemplateValues{
+			Catalog:           "Catalog Title",
+			Component:         "Component Title",
+			AssessmentResults: assessmentResults,
+		},
+	}
+
+	// Run test
+	result, err := CreateTemplateValues(test.catalog, test.assessmentPlan, test.assessmentResults)
+	if err != nil {
```

Comment:
Nit: I noticed that the error is being checked twice (line 181 and 183). Is this intentional?